lib/cli.js

```diff
@@ -278,27 +279,65 @@ exports.process = () => {
 }
 
 exports.run = () => {
-  const config = exports.process()
-
-  switch (config.cmd) {
-    case 'start':
-      new Server(config).start()
-      break
-    case 'run':
-      require('./runner')
-        .run(config)
-        .on('progress', printRunnerProgress)
-      break
-    case 'stop':
-      require('./stopper').stop(config)
-      break
-    case 'init':
-      require('./init').init(config)
-      break
-    case 'completion':
-      require('./completion').completion(config)
-      break
+  const cliOptions = exports.process()
+  const cmd = cliOptions.cmd // prevent config from changing the command
+  const cmdNeedsConfig = cmd === 'start' || cmd === 'run' || cmd === 'stop'
+  let runPromise
+  if (cmdNeedsConfig) {
+    runPromise = cfg.parseConfig(
```

Comment:
I don't like all the noise we have added to this function only for testing. Let's instead make the function `async`, use `await cfg.parseConfig(` on this line and cleanup all other extra code. You don't seem to use any of the return values anyways.

Reply:
I wasn't pleased with it as well. It started to feel like bloat. Using async requires minimal changes and cleans up a number of statements, especially for `init` and `completion`. I'll make the update with the next push.